pkg/provisioner/ironic/ready_test.go

```diff
@@ -102,20 +142,24 @@ func TestProvisionerIsReady(t *testing.T) {
 
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
+			ironicEndpoint := "localhost"
 			if tc.ironic != nil {
 				tc.ironic.Start()
+				ironicEndpoint = tc.ironic.Endpoint()
 				defer tc.ironic.Stop()
 			}
 
+			inspectorEndpoint := "localhost"
```

Comment:
The mock server already provides a default value for the endpoint, even when the pointer is nil, so we don't have to do this sort of thing in every test.

Reply:
Ok